### ISSUE TITLE
Bug: Chat hangs with openai service

### ISSUE THREAD

**Issue by calderonsamuel** [MichelNivard/gptstudio#168]
Sometimes the openai response hangs with the folowing error:

```r
#> Warning: Error in purrr::map: ℹ In index: 1.
#> Caused by error:
#> ! lexical error: invalid char in json text.
#>                                      [DONE]
#>                      (right here) ------^
```

This is because of how we parse the stream. I recently created the `SSEparser` package to help solve this issue. We could import the SSEparser class or create a copy of it to better handle this type of streams.

**Comment by JamesHWade** [comment]
That sounds good to me. Any chance we can get rid of some other dependencies by using your package. 

Keep up the great work! I continue to be impressed by all your contributions.

**Comment by calderonsamuel** [comment]
The fix is ready to be merged! Sorry for the delay, Januay has come with unconventional busy days.

@JamesHWade sadly it wasn't possible to get rid of any dependency, `{SSEparser}` depends on R6, stringr and jsonlite, but we still use all of them for other parts of the app.
@rrodrigowb You can use the development version once is merged, we are working on releasing v0.4.0 soon to CRAN. However I must say that or primarily target users are people working interactively in the Rstudio IDE, so be careful if "in production" means something different for your use case.

**Comment by rrodrigowb** [comment]
@calderonsamuel Thanks for your answer. Finally I have to use another way to implement the streaming feature because I need it also to work async. But your implementation was an inspiration for me.  🙏
